src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -150,50 +177,73 @@ public string GetServerUrl(CommandSettings command)
             return _serverUrl;
         }
 
-        public async Task<int> GetPoolId(CommandSettings command)
+        public async Task<AgentSettings> GetPoolId(CommandSettings command)
         {
-            int poolId;
-
             _projectName = command.GetProjectName(_projectName);
             var deploymentGroupName = command.GetDeploymentGroupName();
 
-            poolId =  await GetPoolIdAsync(_projectName, deploymentGroupName);
-            Trace.Info($"PoolId for deployment group '{deploymentGroupName}' is '{poolId}'.");
-            
-            return poolId;
+            var deploymentGroup =  await GetDeploymentGroupAsync(_projectName, deploymentGroupName);
+            Trace.Info($"PoolId for deployment group '{deploymentGroupName}' is '{deploymentGroup.Pool.Id}'.");
+            Trace.Info($"Project id for deployment group '{deploymentGroupName}' is '{deploymentGroup.Project.Id.ToString()}'.");
+
+            return new AgentSettings() { PoolId = deploymentGroup.Pool.Id, DeploymentGroupId = deploymentGroup.Id, ProjectId = deploymentGroup.Project.Id.ToString()};
```

Comment:
don't new an AgentSettings object, just pass the agentsetting object has parameter and modify it.